.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+      - 2.7
+  pull_request: {}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-old:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest]
+        ghc: ['7.8.4', '7.6.3', '7.4.2']
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      name: Cache cabal stuff
+      with:
+        path: |
+          ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+          dist-newstyle
+        key: ${{ runner.os }}-${{ matrix.ghc }}
+    - name: Install GHC
+      run: |
+        sudo add-apt-repository ppa:hvr/ghc -y
+        sudo apt-get update
+        sudo apt-get install ghc-${{ matrix.ghc }}
+    - name: Build
+      run: |
+        autoreconf -i
+        cabal sdist -z -o .
+        cabal get unix-*.tar.gz
+        cd unix-*/
+        cabal build -w /opt/ghc/bin/ghc-${{ matrix.ghc }} all
+    - name: Haddock
+      run: cabal haddock -w /opt/ghc/bin/ghc-${{ matrix.ghc }}
+
+  build:
+    needs: build-old
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
+        ghc: ['8.10', '8.8', '8.6', '8.4', '8.2', '8.0', '7.10']
```

Comment:
should 7.10 be in build-old?

Reply:
`build-old` is for configurations unsupported by `actions/setup-haskell` (and underlying `ghcup`), which is GHC < 7.10.

Reply:
approved